ui/org.eclipse.pde.bnd.ui/src/org/eclipse/pde/bnd/ui/tasks/AnalyseBundleResolutionJob.java

```diff
@@ -0,0 +1,136 @@
+package org.eclipse.pde.bnd.ui.tasks;
+
+import static java.util.Collections.emptyList;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.bndtools.api.ILogger;
+import org.bndtools.api.Logger;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.pde.bnd.ui.model.resolution.RequirementWrapper;
+import org.osgi.resource.Capability;
+import org.osgi.resource.Namespace;
+
+import aQute.bnd.build.model.EE;
+import aQute.bnd.osgi.resource.ResourceUtils;
+import aQute.lib.io.IO;
+
+public class AnalyseBundleResolutionJob extends Job {
+
+	private static final ILogger					logger	= Logger.getLogger(AnalyseBundleResolutionJob.class);
+
+	private final Set<? extends CapReqLoader>		loaders;
+
+	private Map<String, List<RequirementWrapper>>	requirements;
+	private Map<String, List<Capability>>			capabilities;
+	private EE										ee;
+
+	public AnalyseBundleResolutionJob(String name, Set<? extends CapReqLoader> loaders) {
+		this(name, loaders, null);
+	}
+
+	public AnalyseBundleResolutionJob(String name, Set<? extends CapReqLoader> loaders, EE ee) {
+		super(name);
+		this.loaders = loaders;
+		this.ee = ee;
+	}
+
+	private static <K, V> void mergeMaps(Map<K, List<V>> from, Map<K, List<V>> into) {
+		for (Entry<K, List<V>> entry : from.entrySet()) {
+			K key = entry.getKey();
+
+			List<V> list = into.get(key);
+			if (list == null) {
+				list = new ArrayList<>();
+				into.put(key, list);
+			}
+
+			list.addAll(entry.getValue());
+		}
+	}
+
+	@Override
+	protected IStatus run(IProgressMonitor monitor) {
+		try {
+
+
+			// Load all the capabilities and requirements
+			Map<String, List<Capability>> allCaps = new HashMap<>();
+			Map<String, List<RequirementWrapper>> allReqs = new HashMap<>();
+			for (CapReqLoader loader : loaders) {
+				try {
+					Map<String, List<Capability>> caps = loader.loadCapabilities();
+					mergeMaps(caps, allCaps);
+
+					Map<String, List<RequirementWrapper>> reqs = loader.loadRequirements();
+					mergeMaps(reqs, allReqs);
+				} catch (Exception e) {
+					logger.logError("Error in Bnd resolution analysis.", e);
+				} finally {
+					IO.close(loader);
```

Comment:
use try-with-resources here and maybe ignore the exception (if loader throws one...)